tensorrt_llm/sampling_params.py

```diff
@@ -329,8 +347,11 @@ def _validate(self):
         if self.guided_decoding is not None:
             self.guided_decoding._validate()
 
-        # correct types as users might pass in logprob=True for Top-1 logprobs
-        self.logprobs = self.logprobs and int(self.logprobs)
+        # correct types as users might pass in logprob=True for Top-0 logprobs and logprobs=False for no logprobs
+        if self.logprobs is False:
+            self.logprobs = None
+        if self.logprobs is True:
+            self.logprobs = 0
```

Comment:
I don't understand whats the different meaning of None and 0 here?

Can be just use a typed bool? instead of mixing boolean/None/0-1?